cache/test_h2_cache_control.py

```diff
@@ -1014,3 +1014,442 @@ def test_ignore_max_stale_in_request(self):
+            marks.Param(
+                name="must_revalidate", status="400", r1_hdrs=[("cache-control", "must-revalidate")]
+            ),
+            marks.Param(
+                name="proxy_revalidate",
+                status="400",
+                r1_hdrs=[("cache-control", "proxy-revalidate")],
+            ),
+        ]
+    )
+    def test_use_fresh_on_error(self, name, status, r1_hdrs):
+        """
+        Send first request, this request forwarded to upstream and processed with status
+        code 200, Tempesta cache it. Wait while response become stale. Send second
+        request and receive non stale response, because one of following cache-control
+        parameters is present in message: s-maxage, must-revalidate, proxy-revalidate.
+        """
+        self.use_stale_base("400", "4* 5*", r1_hdrs, [], status, False)
+
+    def test_max_stale(self):
+        """
+        Send first request, this request forwarded to upstream and processed with status
+        code 200, Tempesta cache it. Wait while response become stale. Send second
+        request and receive stale response with status code 200. Wait while
+        mas-stale will expire send another request, receive non stale response.
+        """
+        tempesta = self.get_tempesta()
+        tempesta.config.set_defconfig(tempesta.config.defconfig + f"cache_use_stale 4* 5*;\n")
+        server = self.get_server("deproxy")
+        self.start_all_services(False)
+
+        server.set_response(
+            deproxy.Response.create(
+                status="200",
+                headers=[("Content-Length", "0"), ("cache-control", "max-age=1")],
+                date=deproxy.HttpMessage.date_time_string(),
+            )
+        )
+
+        client = self.get_client("deproxy")
+        client.start()
+        client.send_request(
+            client.create_request(
+                method="GET",
+                uri="/",
+                headers=[],
+            ),
+            "200",
+            3,
+        )
+
+        # Wait while response become stale
+        time.sleep(3)
+
+        server.set_response(
+            deproxy.Response.create(
+                status="400",
+                headers=[("Content-Length", "0")],
+                date=deproxy.HttpMessage.date_time_string(),
+            )
+        )
+
+        client.send_request(
+            client.create_request(
+                method="GET",
+                uri="/",
+                headers=[("cache-control", "max-stale=8")],
+            ),
+            "200",
+            3,
+        )
+
+        resp_age = client.last_response.headers.get("age", -1)
+        # expect stale response
+        self.assertTrue(int(resp_age) >= 0)
+
+        stale_warn = client.last_response.headers.get("warning")
+        self.assertTrue(stale_warn == "110 - Response is stale")
+
+        # Wait while age become greater than max-stale
+        time.sleep(8)
```

Comment:
I would prefer to not change this value, otherwise we risk getting blinking test. Max timeout for first request is 3 then we wait 3 seconds(we don't know how many time previous request is spent) it means, almost 6 seconds in worst case we have before response become stale. Plus another one request with timeout 3, even with value 8 if CI is laggy is possible to blink.